lib/Runtime/Library/JavascriptObject.cpp

```diff
@@ -357,6 +357,32 @@ namespace Js
         return ToStringHelper(args[0], scriptContext);
     }
 
+    Var JavascriptObject::GetToStringTagValue(RecyclableObject *thisArg, ScriptContext *scriptContext)
+    {
+        const PropertyId toStringTagId(PropertyIds::_symbolToStringTag);
+        PolymorphicInlineCache *cache = scriptContext->GetLibrary()->GetToStringTagCache();
+        PropertyValueInfo info;
+        // We don't allow cache resizing, at least for the moment: it's more work, and since there's only one
+        // cache per script context, we can afford to create each cache with the maximum size.
+        PropertyValueInfo::SetCacheInfo(&info, nullptr, cache, false);
+        Var value;
+        if (CacheOperators::TryGetProperty<true, true, true, true, true, true, !PolymorphicInlineCache::IsPolymorphic, PolymorphicInlineCache::IsPolymorphic, false>(
+                thisArg, false, thisArg, toStringTagId, &value, scriptContext, nullptr, &info))
```

Comment:
Filed #3939 about the readability of this template; should probably get inspection from someone more familiar with the caching.

Reply:
Added initializer to JavascriptLibrary ctor; will update PR momentarily.

Yes, I'd love to get feedback on the template parameters.  I'm fairly confident that most of them are correct, but I'm not sure about CheckTypePropertyCache, though I think this should only affect perf, not functional correctness.  @MikeHolman?